kotest-framework/kotest-framework-api/src/commonMain/kotlin/io/kotest/core/internal/TestCaseExecutor.kt

```diff
@@ -78,7 +80,12 @@ class TestCaseExecutor(
    ): TestResult {
       return when {
          extensions.isEmpty() -> executeIfActive(testCase) { executeActiveTest(testCase, context, start) }
-         else -> extensions.first().intercept(testCase) { intercept(it, context, start, extensions.drop(1)) }
+         else -> extensions.first().intercept(testCase) {
```

Comment:
This looks a bit quirky, each lambda created here would always store the entire list of extensions that follow it, even though it only calls the next one. This results in quadratic memory usage over the number of extensions (`n * (n+1) / 2` elements), but could be a lot if there are a lot of extensions, and it is done for each test.

I know it was there before, so it doesn't have anything to do with this change, feel free to ignore it :-)

Consider rewriting the method using a `foldRight` instead of recursing inside the lambda:

```kotlin
      val initialExecute: suspend (TestCase, TestContext) -> TestResult = { testCase, context ->
         executeIfActive(testCase) { executeActiveTest(testCase, context, start) }
      }
      val execute = extensions.foldRight(initialExecute) { extension, execute ->
            { testCase, context ->
               extension.intercept(testCase) {
                  // the user's intercept method is free to change the context of the coroutine
                  // to support this, we should switch the context used by the test case context
                  val newContext = context.withCoroutineContext(coroutineContext)
                  execute(it, newContext)
               }
            }
      }
      return execute(testCase, context)
```

Reply:
This is much nicer.